### PR TITLE
[Feature]: Base32 to C interface

### PR DESCRIPTION
**Pull request by Milerius**
## Description

implements #2120 

Todo: 
 - [x] Base32 C++
 - [x]  Base32 Kotlin
 - [x]  Base32 Swift
 - [x]  Base32 TypeScript

## How to test

- run c++ unit test
- run wasm tests
- run type script tests
- run swift tests

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [ ] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: include/TrustWalletCore/TWBase32.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#include "TWBase.h"
+#include "TWData.h"
+#include "TWString.h"
+
+TW_EXTERN_C_BEGIN
+
+TW_EXPORT_STRUCT
+struct TWBase32;
+
+/**
```
Comment (hewigovens) [inline include/TrustWalletCore/TWBase32.h:18]: We're using `///` style comment, take a look at `TWAES.h`

File: tests/interface/TWBase32Tests.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+#include "Data.h"
+#include <TrustWalletCore/TWBase32.h>
+
+#include <gtest/gtest.h>
+
+TEST(TWBase32, InvalidDecode) {
+    const auto encodedInput = STRING("JBSWY3DPK5XXE3DE=======");
+    auto result = WRAPD(TWBase32Decode(encodedInput.get()));
+    ASSERT_EQ(result, nullptr);
```
Comment (hewigovens) [inline tests/interface/TWBase32Tests.cpp:17]: consider adding some blank line before `ASSERT_EQ` easier to read:
// assign statement
// assign statement

// assert / eq

// another assign / test section
```suggestion`

    ASSERT_EQ(result, nullptr);
```

File: swift/Tests/Base32Tests.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import WalletCore
+
+class Base32Tests: XCTestCase {
+    func testDecode() {
+        if let decoded = Base32.decode(string: "JBSWY3DPK5XXE3DE") {
```
Comment (hewigovens) [inline swift/Tests/Base32Tests.swift:12]: perfect place to use `guard let`

File: swift/Tests/Base32Tests.swift [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            let toCompare = decoded.map({ String(format: "%c", $0) }).joined();
```
Comment (hewigovens) [inline swift/Tests/Base32Tests.swift:13]: I think you can just use `String(data: decoded: encoding:.utf8)`

File: tests/interface/TWBase32Tests.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+TEST(TWBase32, EncodeWithAlphabet) {
+    const auto filecoinAlphabet = STRING("abcdefghijklmnopqrstuvwxyz234567");
+    TW::Data data{'7', 'u', 'o', 'q', '6', 't', 'p', '4', '2', '7', 'u', 'z', 'v', '7', 'f',
+                  'z', 't', 'k', 'b', 's', 'n', 'n', '6', '4', 'i', 'w', 'o', 't', 'f', 'r', 'r', 'i', 's', 't', 'w', 'p', 'r', 'y', 'y'};
+    auto encodedStr = TWBase32EncodeWithAlphabet(&data, filecoinAlphabet.get());
+    std::string result = TWStringUTF8Bytes(encodedStr);
+    ASSERT_EQ(result, "g52w64jworydimrxov5hmn3gpj2gwyttnzxdmndjo5xxiztsojuxg5dxobzhs6i");
+    TWStringDelete(encodedStr);
+}
```
Comment (hewigovens) [inline tests/interface/TWBase32Tests.cpp:57]: Keep last blank line
```suggestion
}

```

**Review by hewigovens — COMMENTED** [review]
left some comments, will check memory issue soon

**Comment by hewigovens** [comment]
I created another issue to track codegen swift code for `Nullable` https://github.com/trustwallet/wallet-core/issues/2368, let's use `Nonnull alphabet` for now

**Review by hewigovens — APPROVED** [review]

**Review by miloserdow — APPROVED** [review]

**Pull request merged**